File: 2015m./uzd1./sprendimai/uzd1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
using namespace std;

vector<int> getTotalGrapesEaten(const vector<int>& grapesEaten) {
    vector<int> grapesEatenCopy = grapesEaten;
    vector<int> updatedGrapesEaten = grapesEaten;
    updatedGrapesEaten.resize(20, 0);

    for (int i = 0; i < 10; ++i) {
        for (int j = i + 1; j < i + 10 - grapesEatenCopy[i] + 1; ++j) {
            updatedGrapesEaten[j]++;
        }
    }
    return updatedGrapesEaten;
}

int main() {
    ifstream data("U1.txt");
    ofstream output("U1rez.txt");

    vector<int> grapesEaten;
    string line;

    int grape;
    while (data >> grape) {
        grapesEaten.push_back(grape);
    }

    vector<int> updatedGrapesEaten = getTotalGrapesEaten(grapesEaten);

    for (int i = 0; i < updatedGrapesEaten.size(); ++i) {
        output << updatedGrapesEaten[i];
        if (i < updatedGrapesEaten.size() - 1) {
            output << " ";
        }
    }

    data.close();
    output.close();

    return 0;
}

File: 2015m./uzd1./sprendimai/uzd1.py
def getTotalGrapesEaten(grapesEaten):
    grapesEatenCopy = grapesEaten
    grapesEaten = grapesEaten + [0]*10

    for i in range(10):
        for j in range(i + 1, i + 10 - grapesEatenCopy[i] + 1):
            grapesEaten[j] += 1
    return grapesEaten

def main():
    data = open("U1.txt", "r")
    output = open("U1rez.txt", "w")

    grapesEaten = next(data).split(" ")
    grapesEaten = [int(o) for o in grapesEaten]

    grapesEaten = getTotalGrapesEaten(grapesEaten)

    grapesEaten = [str(o) for o in grapesEaten]

    output.write(" ".join(grapesEaten))

    data.close()
    output.close()


if __name__ == "__main__":
    main()